.github/workflows/node.js.yml

```diff
@@ -0,0 +1,52 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [main]
+    paths-ignore:
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  general:
+    name: CI run (node ${{ matrix.node-version }})
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [18.x, 20.x, 21.x]
+        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'yarn'
+          cache-dependency-path: ./yarn.lock
+
+      - name: Install dependencies
+        run: yarn --frozen-lockfile
+
+      - name: Building
+        run: yarn build-node
+
+      - name: Formatting and Linting
+        run: yarn format-lint
```

Comment:
This will just run formatting without crashing or failing if the files were not formatted / linted correctly.

Here is an extract from uhttp:

```
        "format": "prettier --write src/ .github/ *.js *.ts *.json *.md",
        "format:ci": "prettier --check src/ .github/ *.js *.ts *.json *.md",
        "lint": "eslint --fix src/",
        "lint:ci": "eslint --max-warnings 0 src/",
```

the `:ci` targets are specifically designed to fail and not to modify any files if linting/formatting is wrong